src/irmin-layers/stats.mli

```diff
@@ -14,65 +14,62 @@
-
-    Note that stats assume that there always is an ongoing freeze. If its not
-    the case, you should discard the last 0.*)
+(** {2 Profiling of Freeze}
 
-val reset_stats : unit -> unit
-val get : unit -> t
+    {3 Control Flow of the Freeze Thread}
+
+    Most of the functions here should be called from under a single mutex lock. *)
+
+val freeze_start : Mtime.t -> string -> unit
+(** Signals the start of a new freeze given the time at which the freeze process
+    started and the name given to the initial code section. *)
+
+val freeze_section : string -> unit
+(** Signals that the freeze is entering a specific section of the code. *)
+
+val freeze_stop : unit -> unit
+(** Signals the end of an ongoing freeze freeze. *)
+
+val freeze_yield : unit -> unit
+(** Signals that the freeze is cooperatively yielding to other threads. *)
+
+val freeze_yield_end : unit -> unit
+(** Signals that the freeze is given back the control. *)
+
+(** {3 Incrementations of Counters} *)
 
 val copy_contents : unit -> unit
-(** Increments t.copied_contents for the current freeze *)
+(** Increments the number of copied contents for the current freeze. *)
 
 val copy_nodes : unit -> unit
-(** Increments t.copied_nodes for the current freeze *)
+(** Increments the number of copied nodes for the current freeze. *)
 
 val copy_commits : unit -> unit
-(** Increments t.copied_commits for the current freeze *)
+(** Increments the number of copied commits for the current freeze. *)
 
 val copy_branches : unit -> unit
-(** Increments t.copied_branches for the current freeze *)
-
-val freeze : unit -> unit
-(** Signals the start of a new freeze, and increments t.nb_freeze *)
+(** Increments the number of copied branches for the current freeze. *)
 
 val add : unit -> unit
-(** Increment the number of objects added, do not call this for objects copied
-    during a freeze. *)
+(** Increment the number of objects added by main thread. *)
 
-val get_adds : unit -> int
-(** Get the number of objects added, not including the copied objects during a
+val skip : unit -> unit
+(** Increment the number of skips during a graph traversal for the current
     freeze. *)
 
-val reset_adds : unit -> unit
-(** Reset the number of objects added. *)
+val copy_newies_loop : unit -> unit
+(** Increment the number of iterations of newies copy. *)
 
-val with_timer : [ `Freeze | `Waiting ] -> (unit -> unit Lwt.t) -> unit Lwt.t
-(** Either the duration of a freeze thread waiting on the freeze lock to start;
-    or the duration of a freeze thread to complete a freeze. *)
+(** {3 Observation} *)
 
-val skip : unit -> unit
-(** Increment the number of skips during a graph traversal. *)
+val get_add_count : unit -> int
+val get_copied_commits_count : unit -> int
+val get_copied_branches_count : unit -> int
+val get_copied_contents_count : unit -> int
+val get_copied_nodes_count : unit -> int
+val get_freeze_count : unit -> int
+val pp_latest : Format.formatter -> unit -> unit
```

Comment:
There is perhaps an extra `unit` here, the declaration doesn't match the implementation. :)

Reply:
Yup, I've just fixed it locally :)